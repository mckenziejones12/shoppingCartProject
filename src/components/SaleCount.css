#inCartContainer {
  background-color: white;
  padding: 3px;
  border-radius: 50%;
  min-height: 20px;
  min-width: 20px;
  color: orange;
  text-align: center;
  align-items: center;
  display: grid;
  align-self: center;
}
