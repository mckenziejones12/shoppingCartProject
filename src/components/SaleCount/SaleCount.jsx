import "./SaleCount.css";

export const SaleCount = ({ count }) => {
  let displayNumberOfItems = count;
  if (count > 9) {
    displayNumberOfItems = "9+";
  }

  return <div id="inCartContainer">{displayNumberOfItems}</div>;
};
