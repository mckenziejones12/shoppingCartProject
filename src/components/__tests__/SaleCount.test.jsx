import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import { SaleCount } from "../SaleCount";

describe("<SaleCount />", () => {
  it("sets item in cart to count 4", () => {
    render(<SaleCount count={4} />);

    expect(screen.getByText("4")).toBeInTheDocument();
  });
  it("sets item in cart to 9+ when 10 or more item in cart", () => {
    render(<SaleCount count={10} />);

    expect(screen.getByText("9+")).toBeInTheDocument();
  });
  it("sets item in cart to 0 when null", () => {
    render(<SaleCount count={null} />);

    expect(screen.getByText("0")).toBeInTheDocument();
  });

  // snapshot
  it("renders correctly", () => {
    const { asFragment } = render(<SaleCount count={3} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
