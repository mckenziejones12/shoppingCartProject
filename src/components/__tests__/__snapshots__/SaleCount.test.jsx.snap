// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SaleCount /> > renders correctly 1`] = `
<DocumentFragment>
  <div
    id="inCartContainer"
  >
    3
  </div>
</DocumentFragment>
`;
